service/configinit.go

```diff
@@ -0,0 +1,530 @@
+	var err error
+	if moduleSource == "auto" {
+		moduleSource, err = findGoModInFileSystem()
+		if err != nil {
+			return err
+		}
+	}
+
+	fileInfo, err := os.Stat(moduleSource)
+	if err != nil {
+		return err
+	}
+	if !fileInfo.IsDir() {
+		return errors.New("module source must be a directory")
+	}
+	goModPath := filepath.Clean(filepath.Join(moduleSource, "go.mod"))
+	file, err := os.ReadFile(goModPath)
+	if err != nil {
+		return err
+	}
+	gomodFile, err := modfile.Parse(goModPath, file, nil)
+	if err != nil {
+		return err
+	}
+	dr = configschema.NewDirResolver(filepath.Dir(gomodFile.Syntax.Name), gomodFile.Module.Mod.Path)
+	return nil
+}
+
+// findGoModInFileSystem attempts to find the go.mod within the repository filesystem
+func findGoModInFileSystem() (string, error) {
+	// if the program was executed using go run, then go.mod exists 2 levels above
+	dirOpt1 := attemptGoMod(filepath.Join("..", "..", "go.mod"))
+	// if the program was executed from the bin folder, but under the repository(created via `make`)
+	dirOpt2 := attemptGoMod("go.mod")
+	if dirOpt1 != "" {
+		return dirOpt1, nil
+	}
+	if dirOpt2 != "" {
+		return dirOpt2, nil
+	}
+	return "", errors.New("cannot determine the module source")
+}
+
+func attemptGoMod(path string) string {
+	rootSrc, _ := filepath.Abs(path)
+	_, err := os.Stat(rootSrc)
+	if err == nil {
+		return filepath.Dir(rootSrc)
+	}
+	return ""
+}
+
+func getAvailableComponentsForPrompts(set *CollectorSettings) *componentsAvailable {
+	c := &componentsAvailable{}
+
+	for _, exp := range set.Factories.Exporters {
+		c.exporters = append(c.exporters, string(exp.Type()))
+	}
+	for _, proc := range set.Factories.Processors {
+		c.processors = append(c.processors, string(proc.Type()))
+	}
+	for _, rec := range set.Factories.Receivers {
+		c.receivers = append(c.receivers, string(rec.Type()))
+	}
+	for _, ext := range set.Factories.Extensions {
+		c.extensions = append(c.extensions, string(ext.Type()))
+	}
+	c.pipelines = []string{string(config.MetricsDataType), string(config.TracesDataType), string(config.LogsDataType)}
+	return c
+}
+
+// showPrompt displays a prompt to the user for configuration options
+func showPrompt(c *componentsAvailable, stdio terminal.Stdio) {
+	promptExp := &survey.MultiSelect{
+		Message: "Select exporters",
+		Options: c.exporters,
+		Filter:  filterFunc,
+	}
+	promptProc := &survey.MultiSelect{
+		Message: "Select processors",
+		Options: c.processors,
+		Filter:  filterFunc,
+	}
+	promptRec := &survey.MultiSelect{
+		Message: "Select receivers",
+		Options: c.receivers,
+		Filter:  filterFunc,
+	}
+	promptExt := &survey.MultiSelect{
+		Message: "Select extensions",
+		Options: c.extensions,
+		Filter:  filterFunc,
+	}
+
+	promptPipelines := &survey.MultiSelect{
+		Message: "Select pipelines to enable",
+		Options: []string{"metrics", "traces", "logs"},
+	}
+
+	promptWithoutComm := &survey.Confirm{
+		Message: "Without commented configuration fields?",
```

Comment:
Do we need this? I think the UX can be simplified by leaving comments in the generated config file and skipping this question.

Reply:
I have removed this option.